### PR TITLE
stateful_pod_control.go: format the code

### PR DESCRIPTION
**Pull request by zhangxiaoyu-zidif**
**What this PR does / why we need it**:

1.Improve the quality of the code.
2.Reduce reduandant parameters
3.add one comma

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by kow3ns** [comment]
Review status: 0 of 1 files reviewed at latest revision, 4 unresolved discussions.

---

*[pkg/controller/statefulset/stateful_pod_control.go, line 122 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/45002#-KiktXryf27Y9z8Y3AbC:-KiktXryf27Y9z8Y3AbD:bpoiqc8) ([raw file](https://github.com/kubernetes/kubernetes/blob/667fc6e083d1dd875963dd374d439a811d595b68/pkg/controller/statefulset/stateful_pod_control.go#L122)):*
> ```Go
> 		attemptedUpdate = true
> 		// commit the update, retrying on conflicts
> 		_, updateErr := spc.client.Core().Pods(set.Namespace).Update(pod)
> ```

Why is this more idiomatic than, or superior to, the current implementation?

---

*[pkg/controller/statefulset/stateful_pod_control.go, line 131 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/45002#-KikrkY1JvUhVfi7bZsk:-KikrkY1JvUhVfi7bZsl:bsmn1wc) ([raw file](https://github.com/kubernetes/kubernetes/blob/667fc6e083d1dd875963dd374d439a811d595b68/pkg/controller/statefulset/stateful_pod_control.go#L131)):*
> ```Go
> 			if copy, err := api.Scheme.DeepCopy(updated); err == nil {
> 				ok := false
> 				pod, ok = copy.(*v1.Pod)
> ```

By convention we don't handle these errors. If DeepCopy does not return a Pod we panic.

---

*[pkg/controller/statefulset/stateful_pod_control.go, line 160 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/45002#-KiktI9VjZi7BslK43Br:-KiktI9VjZi7BslK43Bs:bpoiqbr) ([raw file](https://github.com/kubernetes/kubernetes/blob/667fc6e083d1dd875963dd374d439a811d595b68/pkg/controller/statefulset/stateful_pod_control.go#L160)):*
> ```Go
> 		set.Status.Replicas = replicas
> 		set.Status.ObservedGeneration = &generation
> 		_, updateErr := spc.client.Apps().StatefulSets(set.Namespace).UpdateStatus(set)
> ```

Why is this more idiomatic than, or superior to, the current implementation.

---

*[pkg/controller/statefulset/stateful_pod_control.go, line 168 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/45002#-Kiks6IUrV10j4DdGrsc:-Kiks6IUrV10j4DdGrsd:bkde3be) ([raw file](https://github.com/kubernetes/kubernetes/blob/667fc6e083d1dd875963dd374d439a811d595b68/pkg/controller/statefulset/stateful_pod_control.go#L168)):*
> ```Go
> 			// make a copy so we don't mutate the shared cache
> 			if copy, err := api.Scheme.DeepCopy(updated); err == nil {
> 				ok := false
> ```

As above, by convention we don't handle these errors. If DeepCopy does not return a StatefulSet we panic.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/45002)*

**Comment by kow3ns** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/45002)*

**Comment by zhangxiaoyu-zidif** [comment]
@kow3ns 
Thanks for your review.
1. using 'updateErr" directly can reduce one "err" definition & initialization.
2. using golang comma-ok can prevent panic. If it was not a pod. comma-ok could resolve it, and ok was false, no panic occurs.
Could you approve it now?

File: pkg/controller/statefulset/stateful_pod_control.go [hunk 1]
```diff
@@ -112,23 +112,26 @@ func (spc *realStatefulPodControl) UpdateStatefulPod(set *apps.StatefulSet, pod
...
+		_, updateErr := spc.client.Core().Pods(set.Namespace).Update(pod)
+		if updateErr == nil {
+			return updateErr
 		}
-		updateErr := err
 
 		if updated, err := spc.podLister.Pods(set.Namespace).Get(pod.Name); err == nil {
 			// make a copy so we don't mutate the shared cache
 			if copy, err := api.Scheme.DeepCopy(updated); err == nil {
-				pod = copy.(*v1.Pod)
```
Comment (kargakis) [inline pkg/controller/statefulset/stateful_pod_control.go]: It's unlike that this path will ever fail.

Comment (zhangxiaoyu-zidif) [inline pkg/controller/statefulset/stateful_pod_control.go]: Err，unlike does not mean never，personally speaking，safe code is necessary

Comment (kargakis) [inline pkg/controller/statefulset/stateful_pod_control.go]: Our own code provides guarantees that if violated then we would have a more serious issue than this panic.

Comment (zhangxiaoyu-zidif) [inline pkg/controller/statefulset/stateful_pod_control.go]: @kargakis I will give up these modifications. And as to "updateErr" and "adding comma", do you agree?

Comment (kargakis) [inline pkg/controller/statefulset/stateful_pod_control.go]: Rest of the changes look fine

Comment (zhangxiaoyu-zidif) [inline pkg/controller/statefulset/stateful_pod_control.go]: @kargakis Thanks very much！

**Comment by zhangxiaoyu-zidif** [comment]
@kow3ns @kargakis 
Thanks for your reviewing. Now I withdraw some modifications. 
PTAL

File: pkg/controller/statefulset/stateful_pod_control.go [hunk 2]
```diff
@@ -112,18 +112,17 @@ func (spc *realStatefulPodControl) UpdateStatefulPod(set *apps.StatefulSet, pod
...
 		}
 
 		attemptedUpdate = true
 		// commit the update, retrying on conflicts
-		_, err := spc.client.Core().Pods(set.Namespace).Update(pod)
-		if err == nil {
-			return nil
+		_, updateErr := spc.client.Core().Pods(set.Namespace).Update(pod)
+		if updateErr == nil {
+			return updateErr
```
Comment (kargakis) [inline pkg/controller/statefulset/stateful_pod_control.go]: Just realized that you also changed this one. Please revert it back to nil, when an error returned is nil, it's preferred to explicitly return nil.

Comment (zhangxiaoyu-zidif) [inline pkg/controller/statefulset/stateful_pod_control.go]: OK , that's fine.

**Comment by zhangxiaoyu-zidif** [comment]
@kargakis fix it already. Thanks!
PTAL

**Comment by zhangxiaoyu-zidif** [comment]
@k8s-bot unit test this

**Comment by zhangxiaoyu-zidif** [comment]
@k8s-bot unit test this

**Comment by kargakis** [comment]
@k8s-bot unit test this

**Pull request merged**